add missing get_type() from BlockDevice

pilotak opened the pull request:
according to [this page](https://github.com/ARMmbed/mbed-os/blob/master/docs/design-documents/features/storage/BlockDevices/get_type_method.md) it should have a function `get_type()` but i think this repo should be merged into OS as all other blockdevices are and `I2CEE` return should be added to above page as well

pilotak committed:
add missing get_type() from BlockDevice

pilotak commented:
@geky CI fails because of `mbed-cloud-sdk` which is not relevant to this PR

rdrill commented:
Without this fix under the upgraded mbed-os lib I had an error "Error: Variable type 'I2CEEBlockDevice' is an abstract class".

geky committed:
Fixed travis issues

- Needed new gcc-arm-embedded ppa
- dist: Trusty did not allow apt-get update to work
- test.py is no longer working as it once was
- Needed to update pip versions

geky commented:
Trying to get the CI green but it's being a pain, seems like the last time CI ran was a year ago...

geky commented:
Ok, got it working. Thanks for the PR, sorry it took so long for me to get to it.
